internal/selfmonitor/config/config_builder.go

```diff
@@ -95,6 +95,23 @@ func makeScrapeConfig(scrapeNamespace string) []ScrapeConfig {
 					Action:       Keep,
 					Regex:        "(otelcol_.*|fluentbit_.*|telemetry_.*)",
 				},
+				// The following relabel configs add an artificial pipeline_name label to the Fluent Bit and OTel Collector metrics to simplify pipeline matching
+				// For Fluent Bit metrics, the pipeline_name is based on the name label. Note that a regex group matching Kubernetes resource names (alphanumerical chars and hyphens) is used to extract the pipeline name.
+				// It allows to filter out timeseries with technical names (storage_backend.0, tail.0, etc.)
+				// For OTel Collector metrics, the pipeline_name is extracted from the exporter label, which has the format [otlp|otlphttp]/<pipeline_name>
+				{
+					SourceLabels: []string{"__name__", "name"},
+					Action:       Replace,
+					Regex:        "fluentbit_.+;([a-zA-Z0-9-]+)",
+					TargetLabel:  "pipeline_name",
+				},
+				{
+					SourceLabels: []string{"__name__", "exporter"},
+					Action:       Replace,
+					Regex:        "otelcol_.+;(.+)/([a-zA-Z0-9-]+)",
```

Comment:
We probably could remove the first match group parenthesis to use the default replacement like above

Reply:
Will try it